pkg/controller/jobframework/reconciler.go

```diff
@@ -129,6 +129,18 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		if wl == nil || apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadFinished) {
 			return ctrl.Result{}, nil
 		}
+
+		// if the workload has reclaimable pods, do one last update.
+		if len(wl.Status.ReclaimablePods) > 0 {
+			if rp := job.ReclaimablePods(); !workload.ReclaimablePodsAreEqual(rp, wl.Status.ReclaimablePods) {
+				err = workload.UpdateReclaimablePods(ctx, r.client, wl, rp)
```

Comment:
Actually, let's not do this. It's a separate API call from the condition, which is not desirable.

I would just force the last apply that sets the condition to also set the reclaimable pods.

You can do it in a follow up, but for now, let's not add an extra API call.

Reply:
removed